### ISSUE TITLE
[BUG] 2.8 agents too new for 2.8 server

### ISSUE THREAD

**Issue by tyuhas** [OCSInventory-NG/OCSInventory-ocsreports#1045]
**OCS Inventory version**
Version : 2.8 from RPM (ocsinventory-2.8.0-1.el8.ocs.noarch)

**Describe the bug**
Updating the agents to 2.8 show the following error in the server logs
(From Windows agent)
Fri Oct  9 10:54:28 2020;820115;400;DOVER-2019-06-24-13-43-32;<deleted IP>;OCS-NG_WINDOWS_AGENT_v2.8.0.0;useragent;Bad agent or agent version too recent for server !!
(From Linux agent)
Fri Oct  9 10:14:56 2020;820115;400;webdev-2020-09-15-10-25-13;<deleted IP>;OCS-NG_unified_unix_agent_v2.8.0;useragent;Bad agent or agent version too recent for server !!

**To Reproduce**
1. Use a 2.8 server from the latest OCS repo RPMs
2. Have the 2.8 agent package installed on agent systems
3. Run an inventory from the agent system
4. Check the OCS server logs

**Expected behavior**
(From an older Linux agent)
Fri Oct  9 10:21:10 2020;819901;101;core-2020-09-18-14-47-02;<deleted IP>;OCS-NG_unified_unix_agent_v2.6.0;inventory;transmitted

Note: systems giving the agent error were all using previous versions of the agent and had been connecting to the 2.8 server.  Systems we haven't upgraded agent software are still successfully reporting to the 2.8 server.

**Comment by kapouik** [comment]
Your bug look like a problem on server during the upgrade. 

Look in /usr/share/perl5/vendor_perl/Apache/Ocsinventory.pm
You may find on line 42 this : 
$Apache::Ocsinventory::VERSION = '2.8';

If not, look first if you really get the correct rpm (here is on fedora 32 but you may have similar with 2.8 versions) : 

# rpm -qa |grep -i ocsinventory
ocsinventory-2.8.0-1.fc32.ocs.noarch
ocsinventory-reports-2.8.0-1.fc32.ocs.noarch
ocsinventory-server-2.8.0-1.fc32.ocs.noarch

**Comment by tyuhas** [comment]
Thanks, I was trying to find were that was defined.  In my Ocsinventory.pm it says

$Apache::Ocsinventory::VERSION = '2.8';
$Apache::Ocsinventory::BUILD_VERSION = '760';

My result from the rpm command

ocsinventory-reports-2.8.0-1.el8.ocs.noarch
ocsinventory-2.8.0-1.el8.ocs.noarch
ocsinventory-release-8-1.el8.ocs.noarch
ocsinventory-server-2.8.0-1.el8.ocs.noarch

We're on CentOS 8

**Comment by kapouik** [comment]
Did you have restart your httpd server after upgrading ?

**Comment by tyuhas** [comment]
Yep, multiple times.  Also from the Server Information in the GUI to confirm

PHP Version : 7.3.23
Web Server :  Apache/2.4.37 (centos) OpenSSL/1.1.1c mod_fcgid/2.3.9 mod_perl/2.0.11 Perl/v5.26.3
Database Server : MariaDB Server version 10.3.17-MariaDB
Version OCSReports: 2.8

**Comment by kapouik** [comment]
can you do this command : 

`find /usr/ -name Ocsinventory.pm`

You will get normally one answer : /usr/share/perl5/vendor_perl/Apache/Ocsinventory.pm
If you get another one, like in /usr/local/share, that's the problem.

**Comment by tyuhas** [comment]
That was it!  We did originally use tar-ball installations for the prior OCS versions before migrating to an rpm-based installation.  I just removed everything in the /usr/local/share directory structure but had to resolve a "Can't locate Apache/Ocsinventory/Plugins/Apache.pm" error (for now I copied the Apache.pm that was in the /usr/local/share structure to the /usr/share/perl5/vendor_perl structure).  The agents all now report properly to the server.
